src/main/java/uk/gov/fco/documentupload/service/ocr/OCRService.java

```diff
@@ -0,0 +1,134 @@
+
+@Service
+@Slf4j
+public class OCRService {
+    private boolean enabled;
+
+    private int sharpnessThreshold;
+
+    private RekognitionClient rekognition;
+
+    private TextractClient textract;
+
+    @Autowired
+    public OCRService(@Value("${ocr.enabled}") @NotNull boolean enabled, @Value("${ocr.sharpness.threshold}") int sharpnessThreshold) {
+        this.enabled = enabled;
+        this.sharpnessThreshold = sharpnessThreshold;
+        rekognition = RekognitionClient.builder().region(Region.EU_WEST_2).credentialsProvider(EnvironmentVariableCredentialsProvider.create()).build();
+        textract = TextractClient.builder().region(Region.EU_WEST_2).credentialsProvider(EnvironmentVariableCredentialsProvider.create()).build();
+    }
+
+    public boolean passesQualityCheck(FileUpload upload) throws IOException {
+        if (enabled) {
+            log.info("Starting image quality check");
+            SdkBytes bytes = SdkBytes.fromInputStream(upload.getInputStream());
+            Image image = Image.builder().bytes(bytes).build();
+            DetectLabelsRequest request = DetectLabelsRequest.builder().image(image).featuresWithStrings("IMAGE_PROPERTIES").build();
+            DetectLabelsImageProperties results = rekognition.detectLabels(request).imageProperties();
+            return results.quality().sharpness() >= sharpnessThreshold;
+        }
+        return true;
+    }
+
+    public String extractData(FileUpload upload) throws IOException {
+        if (enabled) {
+            log.info("Starting data extraction");
+            try {
+                SdkBytes bytes = SdkBytes.fromInputStream(upload.getInputStream());
+                Document doc = Document.builder().bytes(bytes).build();
+
+                List<FeatureType> featureTypes = new ArrayList<>();
+                featureTypes.add(FeatureType.QUERIES);
+
+                Query firstNameQuery = Query.builder().alias("firstName").text("Retrieve the subject's first name").build();
+                Query lastNameQuery = Query.builder().alias("lastName").text("Retrieve the subject's surname").build();
+                Query dobQuery = Query.builder().alias("dob").text("Retrieve the user's date of birth").build();
+
+                Collection<Query> queries = new ArrayList<>(Arrays.asList(firstNameQuery, lastNameQuery, dobQuery));
+
+                QueriesConfig queryConfig = QueriesConfig.builder().queries(queries).build();
+
+                AnalyzeDocumentRequest request = AnalyzeDocumentRequest.builder().featureTypes(featureTypes).document(doc).queriesConfig(queryConfig).build();
+
+                AnalyzeDocumentResponse response = textract.analyzeDocument(request);
+
+                List<Block> blocks = response.blocks();
+                List<Block> queryBlocks = this.getBlocksByType(blocks, BlockType.QUERY);
+                List<Block> queryResultBlocks = this.getBlocksByType(blocks, BlockType.QUERY_RESULT);
```

Comment:
you can filter like this:

```suggestion
                List<Block> queryBlocks = blocks.stream().filter(block -> block.blockType() == BlockType.QUERY).collect(Collectors.toList());
```

rather than loop and add to new array